modules/sku/src/ProductManager.php

```diff
@@ -825,42 +825,61 @@ private function updateFields($parent, SKU $sku, array $values) {
     $additionalFields = \Drupal::config('acm_sku.base_field_additions')
       ->getRawData();
 
-    // Loop through all the attributes available for this particular SKU.
-    foreach ($values as $key => $value) {
-      // Check if attribute is required by us.
-      if (isset($additionalFields[$key])) {
-        $field = $additionalFields[$key];
-
-        if ($field['parent'] != $parent) {
-          continue;
-        }
-
-        $field_key = 'attr_' . $key;
-
-        switch ($field['type']) {
-          case 'attribute':
+    // Filter fields for the parent requested.
+    $additionalFields = array_filter($additionalFields, function ($field) use ($parent) {
+      return ($field['parent'] == $parent);
+    });
+
+    // Loop through all the fields we want to read from product data.
+    foreach ($additionalFields as $key => $field) {
+      $source = isset($field['source']) ? $field['source'] : $key;
+
+      // Field key.
+      $field_key = 'attr_' . $key;
+
+      switch ($field['type']) {
+        case 'attribute':
+          // If attribute is not coming in response, then unset it.
+          if (!isset($values[$source])) {
+            $sku->{$field_key}->set(0, NULL);
```

Comment:
any specific reason here to set only 0th delta as null and not the whole value? I think if field has values in 0th and 1st deltas it won't unset the value completely

Reply:
found this while fixing conflict for #43